syntax = "proto3";

package org.couchers.resources;

import "google/protobuf/empty.proto";

service Resources {
  // This is an open service, no authentication is needed

  // This API contains general resources needed by frontends and managed by the backend team

  rpc GetTermsOfService(google.protobuf.Empty) returns (GetTermsOfServiceRes) {
    // Returns the latest Terms of Service markdown, to be shown on signup or if jailed due to this
  }

  rpc GetRegions(google.protobuf.Empty) returns (GetRegionsRes) {
    // Returns a list of allowed regions
  }

  rpc GetLanguages(google.protobuf.Empty) returns (GetLanguagesRes) {
    // Returns a list of allowed languages. This is a subset of the full ISO 639-3 table.
  }
}

message GetTermsOfServiceRes {
  string terms_of_service = 1;  // CommonMark without images
}

message GetRegionsRes {
  // region code to English name
  map<string, string> regions = 1;
}

message GetLanguagesRes {
  // ISO 639-3 lowercase language code to English name
  map<string, string> languages = 1;
}
